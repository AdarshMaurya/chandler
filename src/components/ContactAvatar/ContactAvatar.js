import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {View, Text, Image, ViewPropTypes} from 'react-native';
import {getAvatarUrl, getAvatarColor, getInitials} from 'utils/contacts';

import {styles} from './ContactAvatar.styles';

export class ContactAvatar extends PureComponent {
  static propTypes = {
    contact: PropTypes.object.isRequired,
    size: PropTypes.number.isRequired,
    style: ViewPropTypes.style,
  };

  render() {
    const {contact, size, style} = this.props;

    const avatar = getAvatarUrl(contact);
    const sizeStyle = {width: size, height: size};

    if (avatar) {
      return (
        <Image
          source={{uri: getAvatarUrl(contact)}}
          style={[styles.avatar, sizeStyle, style]}
        />
      );
    }

    return (
      <View
        style={[
          styles.avatar,
          sizeStyle,
          style,
          {backgroundColor: getAvatarColor(contact) || 'rgb(112, 149, 18)'},
        ]}>
        <Text style={styles.text}>{getInitials(contact)}</Text>
      </View>
    );
  }
}
