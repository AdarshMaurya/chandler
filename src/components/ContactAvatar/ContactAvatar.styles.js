import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  avatar: {
    borderRadius: 3,
    justifyContent: 'center',
  },
  text: {
    fontSize: 20,
    color: 'white',
    textAlign: 'center',
  },
});
