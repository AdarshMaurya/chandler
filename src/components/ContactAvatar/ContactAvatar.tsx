import {getAvatarColor, getAvatarUrl, getInitials} from '@utils/contacts';
import React, {PureComponent} from 'react';
import {Image, Text, View, ViewStyle} from 'react-native';

import {IContact} from '@models';
import {styles} from './ContactAvatar.styles';

interface IContactAvatar {
  contact: IContact;
  size: number;
  style?: ViewStyle;
}

export class ContactAvatar extends PureComponent<IContactAvatar> {
  public render() {
    const {contact, size, style} = this.props;

    const avatar = getAvatarUrl(contact);
    const sizeStyle = {width: size, height: size};

    if (avatar) {
      return (
        <Image
          // @ts-ignore TODO Fix this
          source={{uri: getAvatarUrl(contact)}}
          style={[styles.avatar, sizeStyle, style]}
        />
      );
    }

    return (
      <View
        style={[
          styles.avatar,
          sizeStyle,
          style,
          {backgroundColor: getAvatarColor(contact) || 'rgb(112, 149, 18)'},
        ]}
      >
        <Text style={styles.text}>{getInitials(contact)}</Text>
      </View>
    );
  }
}
