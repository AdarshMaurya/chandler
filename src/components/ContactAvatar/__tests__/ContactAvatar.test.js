import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {ContactAvatar} from '../ContactAvatar';

let mockNoAvatar = false;
let mockNoAvatarColor = false;

jest.mock('utils/contacts', () => ({
  getAvatarUrl: () => (!mockNoAvatar ? 'url' : null),
  getAvatarColor: () => (!mockNoAvatarColor ? 'blue' : null),
  getInitials: () => 'TM',
}));

describe('Components', () => {
  describe('ContactAvatar', () => {
    it('should renders correctly', () => {
      expect(
        toJson(shallow(<ContactAvatar contact={{}} size={15} />)),
      ).toMatchSnapshot();
      expect(
        toJson(shallow(<ContactAvatar contact={{}} size={25} />)),
      ).toMatchSnapshot();

      mockNoAvatar = true;
      expect(
        toJson(shallow(<ContactAvatar contact={{}} size={25} />)),
      ).toMatchSnapshot();

      mockNoAvatarColor = true;
      expect(
        toJson(shallow(<ContactAvatar contact={{}} size={25} />)),
      ).toMatchSnapshot();
    });
  });
});
