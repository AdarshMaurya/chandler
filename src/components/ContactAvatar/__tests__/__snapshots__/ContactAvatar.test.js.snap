// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components ContactAvatar should renders correctly 1`] = `
<Image
  source={
    Object {
      "uri": "url",
    }
  }
  style={
    Array [
      Object {
        "borderRadius": 3,
        "justifyContent": "center",
      },
      Object {
        "height": 15,
        "width": 15,
      },
      undefined,
    ]
  }
/>
`;

exports[`Components ContactAvatar should renders correctly 2`] = `
<Image
  source={
    Object {
      "uri": "url",
    }
  }
  style={
    Array [
      Object {
        "borderRadius": 3,
        "justifyContent": "center",
      },
      Object {
        "height": 25,
        "width": 25,
      },
      undefined,
    ]
  }
/>
`;

exports[`Components ContactAvatar should renders correctly 3`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 3,
        "justifyContent": "center",
      },
      Object {
        "height": 25,
        "width": 25,
      },
      undefined,
      Object {
        "backgroundColor": "blue",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "textAlign": "center",
      }
    }
  >
    TM
  </Text>
</View>
`;

exports[`Components ContactAvatar should renders correctly 4`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 3,
        "justifyContent": "center",
      },
      Object {
        "height": 25,
        "width": 25,
      },
      undefined,
      Object {
        "backgroundColor": "rgb(112, 149, 18)",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "textAlign": "center",
      }
    }
  >
    TM
  </Text>
</View>
`;
